Const check for enum discriminants

### Summary

gccrs does not perform a const check for enum discriminants.

### Reproducer

I tried this code:

```rust
fn test() -> isize {
    return 1;
}

fn main() {
    enum Foo {
        Bar = test(),
    }
}
```


### Does the code make use of any (1.49) nightly feature ?

- [ ] Nightly

### Godbolt link

_No response_

### Actual behavior

compiled successfully

### Expected behavior

```
error[E0015]: cannot call non-const function `test` in constants
 --> <source>:7:15
  |
7 |         Bar = test(),
  |               ^^^^^^
  |
  = note: calls in constants are limited to constant functions, tuple structs and tuple variants
```

### GCC Version

64567260de3dd8350cb92ac20a334d0a65ccf316

--- a/gcc/rust/checks/errors/rust-const-checker.cc
+++ b/gcc/rust/checks/errors/rust-const-checker.cc
@@ -646,6 +646,9 @@ ConstChecker::visit (Enum &enum_item)
 {
   check_default_const_generics (enum_item.get_generic_params (),
 				ConstGenericCtx::Enum);
+
+  for (auto &item : enum_item.get_variants ())
+    item->accept_vis (*this);
 }
 
 void

--- a/gcc/rust/resolve/rust-ast-resolve-item.cc
+++ b/gcc/rust/resolve/rust-ast-resolve-item.cc
@@ -356,6 +356,8 @@ ResolveItem::visit (AST::EnumItemDiscriminant &item)
   auto cpath = canonical_prefix.append (decl);
 
   mappings.insert_canonical_path (item.get_node_id (), cpath);
+
+  ResolveExpr::go (item.get_expr (), path, cpath);
 }
 
 void

--- a/gcc/testsuite/rust/compile/enum_discriminant2.rs
+++ b/gcc/testsuite/rust/compile/enum_discriminant2.rs
@@ -1,0 +1,9 @@
+fn test() -> isize {
+    1
+}
+
+enum Foo {
+    Bar = test() // { dg-error "only functions marked as .const." }
+}
+
+fn main() {}

